addons/mail/static/tests/suggestion/suggestion.test.js

```diff
@@ -1340,3 +1340,43 @@ test("Mention with @-role trigger one RPC only", async () => {
         "/web/dataset/call_kw/res.partner/get_mention_suggestions_from_channel",
     ]);
 });
+
+test("Mentioning @everyone with more than 50 members shows warning dialog", async () => {
+    const pyEnv = await startServer();
+    const partnerIds = pyEnv["res.partner"].create(
+        Array(MENTION_AMOUNT_WARNING + 1).fill({ name: "" })
+    );
+    const channelId = pyEnv["discuss.channel"].create({
+        name: "General",
+        channel_type: "channel",
+        channel_member_ids: partnerIds.map((id) => Command.create({ partner_id: id })),
+    });
+    await start();
+    await openDiscuss(channelId);
+    await insertText(".o-mail-Composer-input", "@everyone");
+    await click(".o-mail-Composer-suggestion:contains(Everyone)");
+    await click(".o-mail-ActionList-button[title='Send']");
+    await contains(".modal-body", {
+        text: "You're about to notify 51 users with @everyone. Do you want to continue?",
+    });
+    await click(".modal-footer button", { text: "Yeah, send it!" });
+    await contains(".o-mail-Message", { values: "@everyone" });
```

Comment:
I feel like testing the mention is actually skipped when canceling is important enough to test, especially since we had missing returns in the past that broke features such as [this one](https://github.com/odoo/odoo/pull/237318).